viya4-iac-aws-darksite/custom-ami/README.md

```diff
@@ -0,0 +1,159 @@
+
+![](img/img1.png)
+
+It's an AMI owned by Canonical AWS account (Canonical is company behind Ubuntu) that provides an Ubuntu AWS AMI images.  So you can find that image in the AMI Catalog by searching with the same search criteria as specified in that main.tf file.
+
+![](img/img2.png) ![](img/img3.png)
+
+
+And exactly like in the main.tf (most_recent = true) select the most recent version and click on Launch Instance with AMI.
+
+![](img/img4.png)
+
+
+
+Give the new instance a name that you will remember.  I choosed m5.xlarge as instance type to stay in sync with the iac scripts.
+
+![](img/img5.png)
+
+For the keypair, you'll want to use a keypair you own and the same keypair you plan on using for viya4-iac-aws and viya4-deployment.  If the public key you'd like to use isn't in EC2 already.. import it now (EC2 > key pairs > import key pair).  In the screenshot below, I've already added it named "aws-key".
+
+<img src="img/img6.png" style="zoom:60%;" />
+
+Click launch instance.
+
+![](img/img6b.png)
+
+Wait for the initialization, check for the assigned IP address and setup SSH session to the running EC2-instance with your private SSH-key.  User to use will be ubuntu. 
+
+![](img/img7.png)
+
+
+
+## Install nfs-kernel-server on running EC2-instance
+
+Run the following commands on the Ubuntu EC2-instance.
+
+```shell
+sudo apt update
+
+sudo apt upgrade
+
+sudo apt install nfs-kernel-server
+```
+
+![](img/img8.png)
+
+
+
+![](img/img9.png)
+
+![](img/img10.png)
+
+
+
+You can now stop the EC2-instance.
+
+![](img/img11.png)
+
+
+
+## Create AMI AWS base image 
+
+Once the EC2-instance where you install nfs-kernel-server is stopped, you can create from that EC2-instance a AWS AMI image.  For that click right on the instance, select Image and templates, Create image.
+
+![](img/img12.png)
+
+
+
+Give the image a menaningful name and description and click Create image button.
+
+![](img/img13.png)
+
+![](img/img14.png)
+
+You can find your image here:
+
+![](img/img15.png)
+
+
+
+Search for private images. Initially it will have status pending.
+
+
+
+![](img/img16.png)
+
+
+
+Wait till it become available.
+
+
+
+![](img/img16b.png)
+
+
+
+## Clone viya4-iac-aws and mod so it's using private owned AMI as AWS base image for NFS and JUMP server
+
+I've provided a [helper script](https://gitlab.sas.com/jocobu/viya4-aws-darksite/-/tree/main/viya4-iac-aws/darksite-iac-aws-mods/darksite-iac-aws-mods.sh) to automatically mod the viya4-iac-aws clone.  However if you'd like to do this manually, here are the proceedures:
+
+
+Clone viya4-iac-aws
+
+https://github.com/sassoftware/viya4-iac-aws.git
+
+```shell
+git clone https://github.com/sassoftware/viya4-iac-aws.git
+```
+
+<img src="img/img17.png" style="zoom:60%;" />
```

Comment:
Again, not sure this is the right image, but the `data` block for the image assignment should be text that can be copied if possible.